packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -6688,31 +6688,41 @@ abstract class RenderObjectElement extends Element {
   }
 
   void _updateParentData(ParentDataWidget<ParentData> parentDataWidget) {
-    bool applyParentData = true;
-    assert(() {
-      try {
-        if (!parentDataWidget.debugIsValidRenderObject(renderObject)) {
-          applyParentData = false;
+    if (parentDataWidget.debugIsValidRenderObject(renderObject)) {
+      parentDataWidget.applyParentData(renderObject);
+    } else {
+      // If it's not a valid render object, then don't show the ErrorWidget, or
+      // throw an exception, just report the error and move on.
+      //
+      // This prevents it from crashing in release mode and still reports the
+      // original error in debug mode.
+      assert(() {
+        try {
+          String errorMessage = "Incorrect use of ParentDataWidget.\n";
+
+          if (parentDataWidget.toStringShort() == 'Positioned') {
```

Comment:
Why do we special case the Positioned/Stack case here? There might be an unknown amount of possibilities that lead to the "Incorrect use of ParentDataWidget" error.

Reply:
I remember when I was new to Flutter, the `Incorrect use of ParentDataWidget.` was very confusing to me, so I wanted to have a more descriptive error message as well as to fix the CI tests.

I'll add specific cases only for `Positioned`, `Flexible`, and `Expanded`, leaving the generic error if it doesn't match.